### ISSUE TITLE
Unity 2022.3.23f1 Inspectorでインターフェースクラスがエラーで表示されない問題

### ISSUE THREAD

**Issue by KiyoK7227** [AnnulusGames/Alchemy#82]
Unityのバージョンを2024/04/05にReleaseされた Unity 2022.3.23f1 にアップデートしたところ、
以下のエラーがコンソールに表示され、Inspectorにインターフェースクラスなどが表示できない問題が発生しております。
おそらく、ReleaseNoteを見る限りBuild Systemになにか変更が入ったようなので、そちらが原因かもしれません。

お手数ですが、対応などを検討していただければ幸いです。

```
TargetParameterCountException: Number of parameters specified does not match the expected number.
System.Reflection.RuntimeMethodInfo.ConvertValues (System.Reflection.Binder binder, System.Object[] args, System.Reflection.ParameterInfo[] pinfo, System.Globalization.CultureInfo culture, System.Reflection.BindingFlags invokeAttr) (at <b11ba2a8fbf24f219f7cc98532a11304>:0)
System.Reflection.RuntimeMethodInfo.Invoke (System.Object obj, System.Reflection.BindingFlags invokeAttr, System.Reflection.Binder binder, System.Object[] parameters, System.Globalization.CultureInfo culture) (at <b11ba2a8fbf24f219f7cc98532a11304>:0)
System.Reflection.MethodBase.Invoke (System.Object obj, System.Object[] parameters) (at <b11ba2a8fbf24f219f7cc98532a11304>:0)
Alchemy.Editor.InternalAPIHelper.GetDrawerTypeForType (System.Type classType) (at ./Library/PackageCache/com.annulusgames.alchemy@d874e72f18/Editor/Internal/InternalAPIHelper.cs:29)
Alchemy.Editor.InspectorHelper.BuildElements (UnityEditor.SerializedObject serializedObject, UnityEngine.UIElements.VisualElement rootElement, System.Object target, System.Func`2[T,TResult] findPropertyFunc, System.Int32 depth) (at ./Library/PackageCache/com.annulusgames.alchemy@d874e72f18/Editor/Internal/InspectorHelper.cs:126)
Alchemy.Editor.AlchemyEditor.CreateInspectorGUI () (at ./Library/PackageCache/com.annulusgames.alchemy@d874e72f18/Editor/AlchemyEditor.cs:95)
UnityEditor.UIElements.InspectorElement.CreateInspectorElementUsingUIToolkit (UnityEditor.Editor targetEditor) (at <1899c5bd2a6d466081cd705bee01d689>:0)
UnityEditor.UIElements.InspectorElement.CreateInspectorElementFromSerializedObject (UnityEditor.SerializedObject bindObject) (at <1899c5bd2a6d466081cd705bee01d689>:0)
UnityEditor.UIElements.InspectorElement.ExecuteDefaultActionAtTarget (UnityEngine.UIElements.EventBase evt) (at <1899c5bd2a6d466081cd705bee01d689>:0)
UnityEngine.UIElements.CallbackEventHandler.HandleEvent (UnityEngine.UIElements.EventBase evt) (at <79a7b2c7005043d7ba49637ce6eae1bd>:0)
UnityEngine.UIElements.CallbackEventHandler.HandleEventAtCurrentTargetAndPhase (UnityEngine.UIElements.EventBase evt) (at <79a7b2c7005043d7ba49637ce6eae1bd>:0)
UnityEngine.UIElements.CallbackEventHandler.HandleEventAtTargetPhase (UnityEngine.UIElements.EventBase evt) (at <79a7b2c7005043d7ba49637ce6eae1bd>:0)
UnityEngine.UIElements.CallbackEventHandler.HandleEventAtTargetAndDefaultPhase (UnityEngine.UIElements.EventBase evt) (at <79a7b2c7005043d7ba49637ce6eae1bd>:0)
UnityEditor.UIElements.Bindings.SerializedObjectBindingContext.SendBindingEvent[TEventType] (TEventType evt, UnityEngine.UIElements.VisualElement target) (at <70d92763e0394a19a8eb6aa7c1a65449>:0)
UnityEditor.UIElements.Bindings.SerializedObjectBindingContext.BindTree (UnityEngine.UIElements.VisualElement element, UnityEditor.SerializedProperty parentProperty) (at <70d92763e0394a19a8eb6aa7c1a65449>:0)
UnityEditor.UIElements.Bindings.SerializedObjectBindingContext.ContinueBinding (UnityEngine.UIElements.VisualElement element, UnityEditor.SerializedProperty parentProperty) (at <70d92763e0394a19a8eb6aa7c1a65449>:0)
UnityEditor.UIElements.Bindings.SerializedObjectBindingContext.Bind (UnityEngine.UIElements.VisualElement element) (at <70d92763e0394a19a8eb6aa7c1a65449>:0)
UnityEditor.UIElements.Bindings.DefaultSerializedObjectBindingImplementation.Bind (UnityEngine.UIElements.VisualElement element, UnityEditor.SerializedObject obj) (at <70d92763e0394a19a8eb6aa7c1a65449>:0)
UnityEditor.UIElements.BindingExtensions.Bind (UnityEngine.UIElements.VisualElement element, UnityEditor.SerializedObject obj) (at <1899c5bd2a6d466081cd705bee01d689>:0)
UnityEditor.UIElements.InspectorElement..ctor (UnityEditor.SerializedObject obj, UnityEditor.Editor editor, UnityEditor.UIElements.InspectorElement+DefaultInspectorFramework defaultInspectorFramework) (at <1899c5bd2a6d466081cd705bee01d689>:0)
UnityEditor.UIElements.InspectorElement..ctor (UnityEditor.Editor editor, UnityEditor.UIElements.InspectorElement+DefaultInspectorFramework defaultInspectorFramework) (at <1899c5bd2a6d466081cd705bee01d689>:0)
UnityEditor.UIElements.InspectorElement..ctor (UnityEditor.Editor editor) (at <1899c5bd2a6d466081cd705bee01d689>:0)
UnityEditor.UIElements.EditorElement.BuildInspectorElement () (at <1899c5bd2a6d466081cd705bee01d689>:0)
UnityEditor.UIElements.EditorElement.CreateInspectorElement () (at <1899c5bd2a6d466081cd705bee01d689>:0)
UnityEditor.EditorElementUpdater.CreateInspectorElementsForViewport (UnityEngine.UIElements.ScrollView viewport, UnityEngine.UIElements.VisualElement contentContainer) (at <1899c5bd2a6d466081cd705bee01d689>:0)
UnityEditor.PropertyEditor.RebuildContentsContainers () (at <1899c5bd2a6d466081cd705bee01d689>:0)
UnityEditor.InspectorWindow.RedrawFromNative () (at <1899c5bd2a6d466081cd705bee01d689>:0)
```

**Comment by yenmoc** [comment]
![image](https://github.com/AnnulusGames/Alchemy/assets/44673303/b05e479c-d34e-4b6b-a2b2-cfb02140dae9)

@KiyoK7227 Perhaps the number of parameters of the GetDrawerTypeForType method has changed because of that

You can try temp fix this by adding arguments when calling the method with reflection

![image](https://github.com/AnnulusGames/Alchemy/assets/44673303/38ff7c7a-1eaf-4027-ad4c-4419b726bb3c)

**Comment by KiyoK7227** [comment]
Thanks for the reply.
I appreciate the advice.
I tried adding the function arguments you pointed out to InternalAPIHelper.cs, but the error still did not go away.
Restarting Unity did not help. Is there something else wrong?

**Comment by yenmoc** [comment]
So you can check your unity version to see what parameters the GetDrawerTypeForType function has, then fill in the correct number of parameters. The version I checked here is Unity6.

**Comment by KiyoK7227** [comment]
Thanks for the solution tips.

In InternalAPIHelper.cs, I added only one argument to the following function, and the error no longer occurs.
✕: { classType, null, false }　〇: { classType, null }
However, I don't think this is a fundamental solution, as other problems can still occur.
Please help me to verify the fix.

```
public static Type GetDrawerTypeForType(Type classType)
{
   var instance = EditorAssembly.CreateInstance(Name_ScriptAttributeUtility);
   var utilityType = instance.GetType();

   var bindingFlags = BindingFlags.NonPublic | BindingFlags.Static;
   var methodInfo = utilityType.GetMethod(nameof(GetDrawerTypeForType), bindingFlags);

   return (Type)methodInfo.Invoke(instance, new object[] { classType, null });
}
```

**Comment by yenmoc** [comment]
We need to isolate the process with #if and pass the appropriate arguments with reflection.

**Comment by AnnulusGames** [comment]
This issue was fixed in #88.
